lang/cs/Org.Apache.REEF.Examples/MachineLearning/KMeans/KMeansDriverHandlers.cs

```diff
@@ -79,15 +80,23 @@ public KMeansDriverHandlers()
                 _executionDirectory); 
 
             _totalEvaluators = _partitionsNumber + 1;
-            _mpiDriver = new MpiDriver(Identifier, Constants.MasterTaskId, _fanOut, new AvroConfigurationSerializer());
-
-            _commGroup = _mpiDriver.NewCommunicationGroup(
-               Constants.KMeansCommunicationGroupName,
-               _totalEvaluators)
-                   .AddBroadcast(Constants.CentroidsBroadcastOperatorName, new BroadcastOperatorSpec<Centroids>(Constants.MasterTaskId, new CentroidsCodec()))
-                   .AddBroadcast(Constants.ControlMessageBroadcastOperatorName, new BroadcastOperatorSpec<ControlMessage>(Constants.MasterTaskId, new ControlMessageCodec()))
-                   .AddReduce(Constants.MeansReduceOperatorName, new ReduceOperatorSpec<ProcessedResults>(Constants.MasterTaskId, new ProcessedResultsCodec(), new KMeansMasterTask.AggregateMeans()))
+
+            IConfiguration mpiDriverConfig = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindStringNamedParam<MpiConfigurationOptions.DriverId>(Identifier)
+                .BindStringNamedParam<MpiConfigurationOptions.MasterTaskId>(Constants.MasterTaskId)
+                .BindStringNamedParam<MpiConfigurationOptions.GroupName>(Constants.KMeansCommunicationGroupName)
+                .BindIntNamedParam<MpiConfigurationOptions.FanOut>(_fanOut.ToString(CultureInfo.InvariantCulture).ToString(CultureInfo.InvariantCulture))
+                .BindIntNamedParam<MpiConfigurationOptions.NumberOfTasks>(_totalEvaluators.ToString())
+                .Build();
+
+            _mpiDriver = TangFactory.GetTang().NewInjector(mpiDriverConfig).GetInstance<MpiDriver>();
```

Comment:
Yes, but why is the instance of `MPIDriver` instantiated here via Tang. Can't you just have it passed into the constructor?